File: actualCombat/kotlin-actualCombat/study-case/src/main/java/cn/like/code/kotlin/case_1/f2.kt
package cn.like.code.kotlin.case_1

/**
 * desc:
 * details:
 * @author:  like
 * @since:   2021/6/12 12:21
 * @email:   980650920@qq.com
 */
fun main() {
    val map = hashMapOf("a" to 1, "b" to 2)


    for ((k, v) in map) {
        println("  key : $k, val : $v")
    }

    val a = "123".apply {
        map { it + 1 }
    }
    println(" a: $a")
    println(1.shr(2))

    val list = arrayListOf(1, 2, 3, 3, 4)

    foo { print(123) }
}

fun foo(bar: Int = 1, f1: () -> Unit) {
    f1()
}

File: actualCombat/jvm-programing-language-actualCombat/scala-study-case/src/main/scala/cn/like/code/case_1/f2.scala
package cn.like.code.case_1

/**
 * 匿名函数
 *
 * @author like 980650920@qq.com
 * @since 2021/6/12 15:51
 */
object f2 {

	def main(args: Array[String]): Unit = {
		f(println)

		doubleNumber((a, b) => a + b)
		doubleNumber(_ + _)

		doubleNumber((_ + _), 11, 22)
	}


	def f(func: String => Unit): Unit = {
		func("123")
	}

	def doubleNumber(op: (Int, Int) => Int): Int = {
		op(1, 2)
	}

	def doubleNumber(op: (Int, Int) => Int, a: Int, b: Int): Int = op(a, b)
}

